File: Python/testProgram.py
import pytest
from Program import CPF, CNPJ, PIS, InscricaoEstadualPR, TituloEleitoralPR, Documento

# CPF
def test_validar_numero_completo_cpf():
    cpf = CPF("123456789")
    assert cpf.GetNumeroCompletoCPF() == "12345678909"
    cpf = CPF("1234567")  # Preenchimento de zeros
    assert cpf.GetNumeroCompletoCPF() == "00123456797"

def test_validar_numero_formatado_cpf():
    cpf = CPF("123456789")
    assert Documento.GetNumeroFormatadoCPF(cpf.GetNumeroCompletoCPF()) == "123.456.789-09"
    cpf = CPF("1234567")  # Preenchimento de zeros
    assert Documento.GetNumeroFormatadoCPF(cpf.GetNumeroCompletoCPF()) == "001.234.567-97"

def test_cpf_valido():
    assert CPF.ValidarCPF("12345678909") is True
    assert CPF.ValidarCPF("123.456.789-09") is True

def test_cpf_invalido():
    assert not CPF.ValidarCPF("12345678967")

def test_cpf_invalido_tamanho():
    assert not CPF.ValidarCPF("123456789")

def test_cpf_invalido_caracter():
    assert not CPF.ValidarCPF("123456789xx")

# CNPJ
def test_validar_numero_completo_cnpj():
    cnpj = CNPJ("123456780001")
    assert cnpj.GetNumeroCompletoCNPJ() == "12345678000195"
    cnpj = CNPJ("12345678")  # Preenchimento de zeros
    assert cnpj.GetNumeroCompletoCNPJ() == "00001234567897"

def test_validar_numero_formatado_cnpj():
    cnpj = CNPJ("123456780001")
    assert Documento.GetNumeroFormatadoCNPJ(cnpj.GetNumeroCompletoCNPJ()) == "12.345.678/0001-95"
    cnpj = CNPJ("12345678")  # Preenchimento de zeros
    assert Documento.GetNumeroFormatadoCNPJ(cnpj.GetNumeroCompletoCNPJ()) == "00.001.234/5678-97"

def test_cnpj_valido():
    assert CNPJ.ValidarCNPJ("12345678000195") is True
    assert CNPJ.ValidarCNPJ("12.345.678/0001-95") is True

def test_cnpj_invalido():
    assert not CNPJ.ValidarCNPJ("12345678000101") 

def test_cnpj_invalido_tamanho():
    assert not CNPJ.ValidarCNPJ("123456780001")

def test_cnpj_invalido_caracter():
    assert not CNPJ.ValidarCNPJ("123456780001xx")

# PIS
def test_validar_numero_completo_PIS():
    pis = PIS("1002723088")
    assert pis.GetNumeroCompletoPIS() == "10027230888"
    pis = PIS("10027230")  # Preenchimento de zeros
    assert pis.GetNumeroCompletoPIS() == "00100272304"

def test_validar_numero_formatado_PIS():
    pis = PIS("1002723088")
    assert Documento.GetNumeroFormatadoPIS(pis.GetNumeroCompletoPIS()) == "1.002.723.088-8"
    pis = PIS("10027230")  # Preenchimento de zeros
    assert Documento.GetNumeroFormatadoPIS(pis.GetNumeroCompletoPIS()) == "0.010.027.230-4"

def test_pis_valido():
    assert PIS.ValidarPIS("10027230888") is True
    assert PIS.ValidarPIS("1.002.723.088-8") is True

def test_pis_invalido():
    assert not PIS.ValidarPIS("10027230885")

def test_pis_invalido_tamanho():
    assert not PIS.ValidarPIS("100272308")

def test_pis_invalido_caracter():
    assert not PIS.ValidarPIS("1002723088x")

# InscricaoEstadualPR
def test_validar_numero_completo_IE():
    ie = InscricaoEstadualPR("12345678")
    assert ie.GetNumeroCompletoIE() == "1234567850"
    ie = InscricaoEstadualPR("123456")  # Preenchimento de zeros
    assert ie.GetNumeroCompletoIE() == "0012345607"

def test_validar_numero_formatado_IE():
    ie = InscricaoEstadualPR("12345678")
    assert Documento.GetNumeroFormatadoIE(ie.GetNumeroCompletoIE()) == "123.45678-50"
    ie = InscricaoEstadualPR("123456")  # Preenchimento de zeros
    assert Documento.GetNumeroFormatadoIE(ie.GetNumeroCompletoIE()) == "001.23456-07"

def test_IE_valido():
    assert InscricaoEstadualPR.ValidarIE("1234567850") is True
    assert InscricaoEstadualPR.ValidarIE("123.45678-50") is True

def test_IE_invalido():
    assert not InscricaoEstadualPR.ValidarIE("1234567890")

def test_IE_invalido_tamanho():
    assert not InscricaoEstadualPR.ValidarIE("1234567")

def test_IE_invalido_caracter():
    assert not InscricaoEstadualPR.ValidarIE("12345678xx")

# TituloEleitoralPR
def test_validar_numero_completoTE():
    titulo = TituloEleitoralPR("1023850106")
    assert titulo.GetNumeroCompletoTE() == "102385010671"
    titulo = TituloEleitoralPR("10238506")  # Preenchimento de zeros
    assert titulo.GetNumeroCompletoTE() == "001023850639"

def test_validar_numero_formatadoTE():
    titulo = TituloEleitoralPR("1023850106")
    assert Documento.GetNumeroFormatadoTE(titulo.GetNumeroCompletoTE()) == "1023.8501.0671"
    titulo = TituloEleitoralPR("10238506")  # Preenchimento de zeros
    assert Documento.GetNumeroFormatadoTE(titulo.GetNumeroCompletoTE()) == "0010.2385.0639"

def test_TE_valido():
    assert TituloEleitoralPR.ValidarTE("102385010671") is True
    assert TituloEleitoralPR.ValidarTE("1023.8501.0671") is True

def test_TE_invalido():
    assert not TituloEleitoralPR.ValidarTE("102385010665")

def test_TE_invalido_tamanho():
    assert not TituloEleitoralPR.ValidarTE("10238106")

def test_TE_invalido_caracter():
    assert not TituloEleitoralPR.ValidarTE("1023850106xx")

if __name__ == "__main__":
    pytest.main()

File: CSharp/testProgram.cs
using NUnit.Framework;
using NUnit.Framework.Constraints;

[TestFixture]
public class Tests
{
    //CPF
    [TestCase("123456789", "12345678909")]
    [TestCase("1234567", "00123456797")] //preenchimento de zeros
    public void test_validar_numero_completo_cpf(string cpf_base, string expectedResult)
    {
        CPF cPF = new CPF(cpf_base);
        string result = cPF.GetNumeroCompletoCPF();
        Assert.That(result, Is.EqualTo(expectedResult));
    }

    [TestCase("123456789", "123.456.789-09")]
    [TestCase("1234567", "001.234.567-97")] //preenchimento de zeros
    public void test_validar_numero_formatado_cpf(string cpfBase, string expectedFormattedNumber)
    {
        CPF cpf = new CPF(cpfBase);
        string result = Documento.GetNumeroFormatadoCPF(cpf.GetNumeroCompletoCPF());
        Assert.That(result, Is.EqualTo(expectedFormattedNumber));
    }

    [TestCase("12345678909", true)]
    [TestCase("123.456.789-09", true)]
    [TestCase("12345678934", false)]
    [TestCase("1234567890a", false)]
    [TestCase("123456789012", false)]
    public void test_valida_cpf(string cpfCompleto, bool expectedResult)
    {
        bool result = CPF.ValidarCPF(cpfCompleto);
        Assert.That(result, Is.EqualTo(expectedResult));
    }

    //CNPJ
    [TestCase("123456780001", "12345678000195")]
    [TestCase("12345678", "00001234567897")] //preenchimento de zeros
    public void test_validar_numero_completo_cnpj(string cnpj_base, string expectedResult)
    {
        CNPJ cNPJ = new CNPJ(cnpj_base);
        string result = cNPJ.GetNumeroCompletoCNPJ();
        Assert.That(result, Is.EqualTo(expectedResult));
    }

    [TestCase("123456780001", "12.345.678/0001-95")]
    [TestCase("12345678", "00.001.234/5678-97")] //preenchimento de zeros
    public void test_validar_numero_formatado_cnpj(string cnpjBase, string expectedFormattedNumber)
    {
        CNPJ cnpj = new CNPJ(cnpjBase);
        string result = Documento.GetNumeroFormatadoCNPJ(cnpj.GetNumeroCompletoCNPJ());
        Assert.That(result, Is.EqualTo(expectedFormattedNumber));
    }

    [TestCase("12345678000195", true)]
    [TestCase("12.345.678/0001-95", true)]
    [TestCase("12345678000156", false)]
    [TestCase("123456780001xx", false)]
    [TestCase("123456780001", false)]
    public void test_valida_cnpj(string cnpjCompleto, bool expectedResult)
    {
        bool result = CNPJ.ValidarCNPJ(cnpjCompleto);
        Assert.That(result, Is.EqualTo(expectedResult));
    }


    //PIS
    [TestCase("1002723088", "10027230888")]
    [TestCase("10027230", "00100272304")] //preenchimento de zeros
    public void test_validar_numero_completo_pis(string pis_base, string expectedResult)
    {
        PIS pIS = new PIS(pis_base);
        string result = pIS.GetNumeroCompletoPIS();
        Assert.That(result, Is.EqualTo(expectedResult));
    }

    [TestCase("1002723088", "1.002.723.088-8")]
    [TestCase("10027230", "0.010.027.230-4")] //preenchimento de zeros
    public void test_validar_numero_formatado_PIS(string pisBase, string expectedFormattedNumber)
    {
        PIS pis = new PIS(pisBase);
        string result = Documento.GetNumeroFormatadoPIS(pis.GetNumeroCompletoPIS());
        Assert.That(result, Is.EqualTo(expectedFormattedNumber));
    }

    [TestCase("10027230888", true)]
    [TestCase("1.002.723.088-8", true)]
    [TestCase("10027230884", false)]
    [TestCase("1002723088x", false)]
    [TestCase("100272308", false)]
    public void test_valida_pis(string pisCompleto, bool expectedResult)
    {
        bool result = PIS.ValidarPIS(pisCompleto);
        Assert.That(result, Is.EqualTo(expectedResult));
    }


    //IE
    [TestCase("12345678", "1234567850")]
    [TestCase("123456", "0012345607")] //preenchimento de zeros
    public void test_validar_numero_completo_IE(string ie_base, string expectedResult)
    {
        InscricaoEstadualPR inscricaoEstadualPR = new InscricaoEstadualPR(ie_base);
        string result = inscricaoEstadualPR.GetNumeroCompletoIE();
        Assert.That(result, Is.EqualTo(expectedResult));
    }

    [TestCase("12345678", "123.45678-50")]
    [TestCase("123456", "001.23456-07")] //preenchimento de zeros
    public void test_validar_numero_formatado_IE(string ieBase, string expectedFormattedNumber)
    {
        InscricaoEstadualPR inscricaoEstadualPR = new InscricaoEstadualPR(ieBase);
        string result = Documento.GetNumeroFormatadoIE(inscricaoEstadualPR.GetNumeroCompletoIE());
        Assert.That(result, Is.EqualTo(expectedFormattedNumber));
    }

    [TestCase("1234567850", true)]
    [TestCase("123.45678-50", true)]
    [TestCase("1234567829", false)]
    [TestCase("12345678xx", false)]
    [TestCase("1234567", false)]
    public void test_valida_IE(string ieCompleto, bool expectedResult)
    {
        bool result = InscricaoEstadualPR.ValidarIE(ieCompleto);
        Assert.That(result, Is.EqualTo(expectedResult));
    }


    //TE
    [TestCase("1023850106", "102385010671")]
    [TestCase("10238506", "001023850639")] //preenchimento de zeros
    public void test_validar_numero_completo_TE(string te_base, string expectedResult)
    {
        TituloEleitoralPR tituloEleitoralPR = new TituloEleitoralPR(te_base);
        string result = tituloEleitoralPR.GetNumeroCompletoTE();
        Assert.That(result, Is.EqualTo(expectedResult));
    }

    [TestCase("102385010671", "1023.8501.0671")]
    [TestCase("001023850639", "0010.2385.0639")]
    public void test_validar_numero_formatado_TE(string teBase, string expectedFormattedNumber)
    {
        TituloEleitoralPR tituloEleitoralPR = new TituloEleitoralPR(teBase);
        string result = Documento.GetNumeroFormatadoTE(tituloEleitoralPR.GetNumeroCompletoTE());
        Assert.That(result, Is.EqualTo(expectedFormattedNumber));
    }

    [TestCase("102385010671", true)]
    [TestCase("1023.8501.0671", true)]
    [TestCase("102385010689", false)]
    [TestCase("1023850106xx", false)]
    [TestCase("10238506", false)]
    public void test_valida_TE(string teCompleto, bool expectedResult)
    {
        bool result = TituloEleitoralPR.ValidarTE(teCompleto);
        Assert.That(result, Is.EqualTo(expectedResult));
    }

}
